yt_dlp/extractor/tumblr.py

```diff
@@ -107,116 +237,170 @@ class TumblrIE(InfoExtractor):
+                    # there are multiple formats; extract them
+                    # (and ignore other sources of width/height data as they may be wrong)
+                    sources = []
+                    sd_url = self._search_regex(
+                        r'<source[^>]+src=(["\'])(?P<url>.+?)\1', iframe,
+                        'sd video url', default=None, group='url')
+                    if sd_url:
+                        sources.append((sd_url, 'sd'))
+                    sources.append((hd_url, 'hd'))
+
+                    formats = [{
+                        'url': video_url,
+                        'format_id': format_id,
+                        'height': int_or_none(self._search_regex(
+                            r'_(\d+)\.\w+$', video_url, 'height', default=None)),
+                        'quality': quality,
+                    } for quality, (video_url, format_id) in enumerate(sources)]
+
+        if 'url' not in media_json and not video_url and not iframe_url:
+            # external video host (but we weren't able to figure it out from the api)
             iframe_url = self._search_regex(
                 r'src=["\'](https?://safe\.txmblr\.com/svc/embed/inline/[^"\']+)["\']',
                 webpage, 'embed iframe url', default=None)
             return self.url_result(iframe_url or redirect_url, 'Generic')
 
-        iframe = self._download_webpage(
-            iframe_url, video_id, 'Downloading iframe page',
-            headers={'Referer': redirect_url})
-
-        duration = None
-        sources = []
-
-        sd_url = self._search_regex(
-            r'<source[^>]+src=(["\'])(?P<url>.+?)\1', iframe,
-            'sd video url', default=None, group='url')
-        if sd_url:
-            sources.append((sd_url, 'sd'))
-
-        options = self._parse_json(
-            self._search_regex(
-                r'data-crt-options=(["\'])(?P<options>.+?)\1', iframe,
-                'hd video url', default='', group='options'),
-            video_id, fatal=False)
-        if options:
-            duration = int_or_none(options.get('duration'))
-            hd_url = options.get('hdUrl')
-            if hd_url:
-                sources.append((hd_url, 'hd'))
-
-        formats = [{
-            'url': video_url,
-            'ext': 'mp4',
-            'format_id': format_id,
-            'height': int_or_none(self._search_regex(
-                r'/(\d{3,4})$', video_url, 'height', default=None)),
-            'quality': quality,
-        } for quality, (video_url, format_id) in enumerate(sources)]
-
+        formats = formats or [{
+            'url': media_json.get('url', video_url),
+            'format_id': '0',
+            'width': media_json.get('width', int_or_none(
+                self._og_search_property('video:width', webpage, default=None))),
+            'height': media_json.get('height', int_or_none(
+                self._og_search_property('video:height', webpage, default=None))),
```

Comment:
```suggestion
            'width': int_or_none(media_json.get('width') or self._og_search_property('video:width', webpage, default=None)),
            'height': int_or_none(media_json.get('height') or self._og_search_property('height', webpage, default=None)),
```

Reply:
Will do, but @Ashish0804 please see above.

Reply:
The property is `video:height` and not `height`; I have otherwise committed this change as written despite the >120-character lines. (I take it you prefer this to the unnecessary parentheses flake8 will otherwise want?)

Reply:
The line limit is not  a hard rule. It's barely over 120 chars and is fine. But you could also do:
```py
            'width': int_or_none(
                media_json.get('width') or self._og_search_property('video:width', webpage, default=None)),
            'height': int_or_none(
                media_json.get('height') or self._og_search_property('video:height', webpage, default=None)),
```
The point of the suggestion was to wrap the whole thing in `int_or_none` rather than putting _og_search_property inside the `.get`